Kentor.AuthServices/Internal/QueryStringHelper.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Kentor.AuthServices.Internal
+{
+    /// <summary>
+    /// Class implements static methods to help parse a query string.
+    /// </summary>
+    public static class QueryStringHelper
+    {
+        /// <summary>
+        /// Splits a query string into its key/value pairs. 
+        /// </summary>
+        /// <param name="queryString">A query string, with or without the leading '?' character.</param>
+        /// <returns>A collecktion with the parsed keys and values.</returns>
+        public static NameValueCollection ParseQueryString(String queryString)
```

Comment:
Is the old NameValueCollection really the best type to return? Could we go with a more modern IDictionary<string, string>? What is used in Owin? Could we use the same type?